### PR TITLE
PSCE-303 feat: adds trestlebot-sync-upstreams command

### PR DESCRIPTION
**Pull request by jpower432**
## Description

Adds `trestlebot-sync-upstreams` command.

This add the command based on the task created in #137 with initial unit tests and modified E2E test. A follow on PR will add the GitHub Action and documentation on usage.

Stacked with #137 

## Review Hints

Review all of the commits together.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How has this been tested?

- [X] E2E tests added/updated
- [X] Tested with Git/CI provider (GitLab CI)

## Checklist

- [X] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/testutils.py [hunk 1]
```diff
@@ -321,20 +354,74 @@ def build_test_command(
...
+        ]
+    )
+    return command
+
+
+def prepare_upstream_repo() -> str:
+    """
+    Prepare a temporary upstream repo for testing.
+
+    The include the test NIST catalog and a modified profile.
```
Comment (beatrizmcouto) [inline tests/testutils.py:403]: Replace the with this

File: tests/testutils.py [hunk 2]
```diff
@@ -321,20 +354,74 @@ def build_test_command(
...
+    """
+    tmp_dir = pathlib.Path(tempfile.mkdtemp())
+    repo: Repo = repo_setup(tmp_dir)
+    load_from_json(
+        tmp_dir, "simplified_nist_catalog", "simplified_nist_catalog", cat.Catalog
+    )
+
+    # Modify the profile to include an additional control and write it out
+    src_path = JSON_TEST_DATA_PATH / "simplified_nist_profile.json"
```
Comment (beatrizmcouto) [inline tests/testutils.py:412]: Add some additional comments on documentation on the why

Comment (beatrizmcouto) [reply tests/testutils.py:412]: In E2E test documentation

File: tests/trestlebot/entrypoints/test_sync_upstreams.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    assert len(commit.stats.files) == 1
+
+    # Clean up the source repo
+    clean(source, None)
+
+
+def test_with_exclude_models(
+    tmp_repo: Tuple[str, Repo], valid_args_dict: Dict[str, str]
+) -> None:
+    """Test with exclude models."""
```
Comment (beatrizmcouto) [inline tests/trestlebot/entrypoints/test_sync_upstreams.py:98]: Perhaps there could be more detail on exactly what is being tested in the description

File: tests/trestlebot/entrypoints/test_sync_upstreams.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    args_dict["sources"] = f"{source}@main"
+
+    with patch("git.remote.Remote.push") as mock_push, patch(
+        "sys.argv", ["trestlebot", *args_dict_to_list(args_dict)]
+    ):
+        mock_push.return_value = "Mocked Results"
+        with pytest.raises(SystemExit, match="0"):
+            cli_main()
+
+    # Verify the the correct files were included
```
Comment (beatrizmcouto) [inline tests/trestlebot/entrypoints/test_sync_upstreams.py:116]: replace first "the" with "that"

File: trestlebot/entrypoints/sync_upstreams.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        """Run the sync upstreams entrypoint."""
+        exit_code: int = SUCCESS_EXIT_CODE
+        try:
+            set_log_level_from_args(args)
+            if not args.sources:
+                raise EntrypointInvalidArgException(
+                    "--sources", "Must set at least one source to sync from."
+                )
+
+            # Assume that is skip_items is not set, then
```
Comment (beatrizmcouto) [inline trestlebot/entrypoints/sync_upstreams.py:91]: Replace "is" with "if"

File: trestlebot/entrypoints/sync_upstreams.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+            # skip nothing and if include items is not set, then include all.
+            include_model_list: List[str] = ["*"]
+            if args.include_models:
+                include_model_list = comma_sep_to_list(args.include_models)
+            model_filter: ModelFilter = ModelFilter(
+                skip_patterns=comma_sep_to_list(args.exclude_models),
+                include_patterns=include_model_list,
+            )
+
+            # Should be false is skip_validation is true
```
Comment (beatrizmcouto) [inline trestlebot/entrypoints/sync_upstreams.py:101]: Replace is with if

**Review by beatrizmcouto — APPROVED** [review]
Minor text modifications proposed, looks good to me

**Comment by jpower432** [comment]
@beatrizmcouto Updates added. PTAL.

**Comment by jpower432** [comment]
@beatrizmcouto Putting back in draft to revisit the include/exclude flags and descriptions.

**Comment by jpower432** [comment]
@JimFuller-RedHat PTAL

**Review by JimFuller-RedHat — APPROVED** [review]
LGTM

File: trestlebot/entrypoints/sync_upstreams.py [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+            super().run_base(args, pre_tasks)
+        except Exception as e:
+            exit_code = handle_exception(e)
+
+        sys.exit(exit_code)
+
+
+def main() -> None:
+    """Run the CLI."""
+    parser = argparse.ArgumentParser(
```
Comment (JimFuller-RedHat) [inline trestlebot/entrypoints/sync_upstreams.py:120]: just an observation about developing CLI - we have found https://palletsprojects.com/p/click/ to be a higher starting point when developing CLIs... not suggesting to refactor but thought you might be interested

Comment (jpower432) [reply trestlebot/entrypoints/sync_upstreams.py:120]: Thanks for the suggestion! I think `click` could be really useful if/when we decide that we need to start grouping commands for ease and discovererabilty. So far they have been standalone/dissimilar so we have keep it is as a flat structure.

**Pull request merged**